lib/product-import.coffee

```diff
@@ -182,6 +185,70 @@ class ProductImport
   _generateUniqueToken: ->
     _.uniqueId "#{new Date().getTime()}"
 
+  _fetchAndResolveCustomReferences: (product) =>
+    new Promise (resolve) =>
+      if product.masterVariant
+        @_fetchAndResolveCustomReferencesByVariant(product.masterVariant).then (result) -> product.masterVariant = result
+
+      if product.variants
+        Promise.map product.variants, (variant) =>
+          @_fetchAndResolveCustomReferencesByVariant(variant)
+          .then ->
+            Promise.resolve()
+        ,{concurrency: 5}
+      resolve(product)
+
+  _fetchAndResolveCustomReferencesByVariant: (variant) =>
+    new Promise (resolve) =>
+      if variant.attributes
+        _.map variant.attributes, (attribute) =>
+          if attribute and _.isArray(attribute.value)
+            if _.every(attribute.value, @_isReferenceTypeAttribute) # all elements in the attribute array should be a ref.
+              @_resolveCustomReferenceSet(attribute.value)
+              .then (result) ->
+                attribute.value = result
+                resolve(variant)
+          else
+            if attribute and @_isReferenceTypeAttribute(attribute.value)
+              @_resolveCustomReference(attribute.value)
+              .then (result) =>
+                attribute.value = result
+                resolve(variant)
+
+
+  _resolveCustomReferenceSet: (attribute) =>
+    # resolve all references and return a list of resolved values.
+    new Promise (resolve, reject) =>
+      values = []
+      Promise.map attribute, (referenceObject) =>
+        @_resolveCustomReference(referenceObject)
+        .then (result) ->
+          values.push(result)
+          if _.size(values) is _.size(attribute)
+            resolve(values)
+          Promise.resolve()
+        .catch (err) ->
+          reject err
+
+
+  _isReferenceTypeAttribute: (attributeValue) ->
+    _.has(attributeValue, 'resolvePredicate') and _.has(attributeValue, 'endpoint')
```

Comment:
Sorry, don't understand how u mean specific for a customer?
This is to resolve any custom reference present as an attribute. A custom reference attribute can be defined as mentioned here:
https://github.com/sphereio/sphere-product-import/issues/10

It will be good to document this.